cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -222,6 +258,7 @@ metadata:
   namespace: kube-system
   labels:
     k8s-app: kube-dns
+    kubernetes.io/name: "CoreDNS"
```

Comment:
Adding this, current dns pods will be deleted and recreated.

Reply:
and that is also true!
maybe one day if the service is renamed from `kube-dns` we can change this too..

i'm inclined to close this PR for now @ojmhetar 
but feel free to have a look at the kubernetes/kubeadm issue tracker.
there are a number of items with help-wanted there. :)

/close

Reply:
I understand, thank you for taking a look!

Reply:
no problem, feel free to help us in other areas! :)